tests/test_hoist.py

```diff
@@ -0,0 +1,264 @@
+"""
+A selection of tests for the generic hoist functionality.
+"""
+from pathlib import Path
+import pytest
+import numpy as np
+
+from conftest import available_frontends, jit_compile, clean_test
+from loki import FindVariables
+from loki.expression import symbols as sym
+from loki import ir, as_tuple, is_iterable
+from loki import Scheduler
+from loki.transform import HoistAnalysis, HoistSynthesis
+
+write_transformation_output = False
```

Comment:
What's the purpose of this? If it's not going to be used, I would suggest removing it rather than leaving dead code in the tests.